### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ivanzh13**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
+        if person.get("wife"):
+            temp_person.wife = Person.people.get(person["wife"])
+            if temp_person.wife:
+                temp_person.wife.husband = temp_person
+        if person.get("husband"):
+            temp_person.husband = Person.people.get(person["husband"])
+            if temp_person.husband:
+                temp_person.husband.wife = temp_person
+        person_list.append(temp_person)
+    return person_list
```
Comment (ArsenPidhoretskyi) [inline app/main.py:23]: Do this in tow loop and simplify code.
In first loop create all Person
In second loop set wife/husband

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
+        self.age = age
+        self.__class__.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = list()
+    for person in people:
+        _ = Person(person["name"], person["age"])
```
Comment (vasylhnatiuk) [inline app/main.py:13]:
```suggestion
       person_list.append(Person(person["name"], person["age"]))
```

File: app/main.py [hunk 3]
```diff
@@ -1,8 +1,20 @@
 class Person:
-    # write your code here
-    pass
+    people = dict()
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[name] = self
```
Comment (vasylhnatiuk) [inline app/main.py:7]:
```suggestion
        self.people[name] = self
```

File: app/main.py [hunk 4]
```diff
@@ -1,8 +1,20 @@
...
+    person_list = list()
+    for person in people:
+        _ = Person(person["name"], person["age"])
+    
+    for person in people:
+        if person.get("wife"):
+            Person.people.get(person["name"]).wife = Person.people.get(person["wife"])
+        if person.get("husband"):
+            Person.people.get(person["name"]).husband = Person.people.get(person["husband"])
+    return list(Person.people.values())
```
Comment (vasylhnatiuk) [inline app/main.py:20]:
```suggestion
    return person_list
```

File: app/main.py (hunk 2 above)
Comment (vasylhnatiuk) [reply app/main.py:13]: it will be better way to return list

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]
image

File: app/main.py (hunk 3 above)
Comment (ivanzh13) [reply app/main.py:7]: Explain why if we need add to class's attribute we use instance of class not class

Comment (vasylhnatiuk) [reply app/main.py:7]: sorry, my mistake, but it would be better to use Person.people

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

**Review by vasylhnatiuk — APPROVED** [review]

**Pull request closed without merging**